File: programmers_question.cpp
// WAP to store name, company, salary and programming languages of some programmers in c++ and display the result.

#include <iostream>
using namespace std;
#include <list>
class Programmers
{
private:
    string name;
    string company;
    string salary;
    int noLang;
    list<string> langList;

public:
    void getInfo()
    {
        string temp;
        cout << "Enter name : ";
        getline(cin, name);
        cout << "Enter company : ";
        getline(cin, company);
        cout << "Enter salary : ";
        getline(cin, salary);
        cout << "Enter no. of programming languages u know : ";
        cin >> noLang;
        getline(cin, temp);
        for (int i = 1; i <= noLang; i++)
        {
            cout << "Enter language " << i << " : ";
            getline(cin, temp);
            langList.push_back(temp);
        }
    }
    void showInfo()
    {
        cout << "Name : " << name << endl;
        cout << "Company : " << company << endl;
        cout << "Salary : " << salary << endl;
        cout << "Languages : ";
        for (string item : langList)
            cout << item << "  ";
        cout << endl;
    }
};
main()
{
    int n;
    string temp;
    cout << "\033[H\033[2J";
    cout << "Enter the no. of programmers : ";
    cin >> n;
    getline(cin, temp);
    Programmers p[n];
    for (int i = 0; i < n; i++)
    {
        cout << "\nFor programmer " << i + 1 << " :\n";
        p[i].getInfo();
    }
    cout << "Press any key to see stored data...";
    system("pause>0");
    system("cls");
    cout << "********* STORED DATA **********\n";
    for (int i = 0; i < n; i++)
    {
        cout << "\nFor programmer " << i + 1 << " :\n";
        p[i].showInfo();
    }
    cout << "\nPress any key to exit ...";
    system("pause>0");
    cout << "\nExited successfully!";
}

File: ProgrammersQuestion.java
// WAP to store name, company, salary and programming languages of some programmers in java and display the result.

import java.util.Scanner;

class Programmers {
    private String name;
    private String company;
    private int salary;
    private int noLang;
    private String[] langList;

    public Programmers(String Name, String Company, int Salary) {
        name = Name;
        company = Company;
        salary = Salary;
    }

    public void setLangList(int Nolang, String[] Langlist) {
        langList = Langlist;
        noLang = Nolang;
    }

    public void showInfo() {
        System.out.println(String.format("Name : %s\nCompany : %s\nSalary : %d", name, company, salary));
        System.out.print("Languages (" + noLang + ") : ");
        for (int i = 0; i < noLang; i++)
            System.out.print(langList[i] + "  ");
        System.out.println();
    }
}

public class ProgrammersQuestion {
    public static void main(String[] args) {
        System.out.print("\033[H\033[2J");
        Scanner scan = new Scanner(System.in);
        System.out.print("Enter the no. of programmers : ");
        int N = scan.nextInt();
        scan.nextLine();
        Programmers[] pr = new Programmers[N];
        for (int i = 0; i < N; i++) {
            System.out.print("\nFor programmer " + (i + 1) + ":\n");
            System.out.print("Enter name : ");
            String NAME = scan.nextLine();
            System.out.print("Enter company : ");
            String COMPANY = scan.nextLine();
            System.out.print("Enter salary per month in US $ : ");
            int SALARY = scan.nextInt();
            pr[i] = new Programmers(NAME, COMPANY, SALARY);
            System.out.print("Enter the no. of languages u know : ");
            int NOLANG = scan.nextInt();
            scan.nextLine();
            String[] LANGLIST = new String[NOLANG];
            for (int j = 0; j < NOLANG; j++) {
                System.out.print("Enter language " + (j + 1) + " : ");
                LANGLIST[j] = scan.nextLine();
            }
            pr[i].setLangList(NOLANG, LANGLIST);
        }
        System.out.print("\nPress 'ENTER' key to see stored data...");
        scan.nextLine();
        System.out.print("\033[H\033[2J");
        System.out.println("********** STORED DATA *********");
        for (int i = 0; i < N; i++) {
            System.out.println("\nFor programmer " + (i + 1) + " :");
            pr[i].showInfo();
        }
        System.out.print("\nPress 'ENTER' to exit...");
        scan.nextLine();
        scan.close();
        System.out.print("Exited sucessfully.");
    }
}
